src/drivers/device/nuttx/I2C.cpp

```diff
@@ -199,17 +180,17 @@ I2C::transfer(const uint8_t *send, unsigned send_len, uint8_t *recv, unsigned re
 		msgs = 0;
 
 		if (send_len > 0) {
-			msgv[msgs].frequency = _bus_clocks[_bus - 1];
-			msgv[msgs].addr = _address;
+			msgv[msgs].frequency = _bus_clocks[get_device_bus() - 1];
+			msgv[msgs].addr = get_device_address();
 			msgv[msgs].flags = 0;
 			msgv[msgs].buffer = const_cast<uint8_t *>(send);
 			msgv[msgs].length = send_len;
 			msgs++;
 		}
 
 		if (recv_len > 0) {
-			msgv[msgs].frequency = _bus_clocks[_bus - 1];;
-			msgv[msgs].addr = _address;
+			msgv[msgs].frequency = _bus_clocks[get_device_bus() - 1];;
```

Comment:
On stm32 bus numbers are 1 based, on other archs they are 0 based. This should be a get-er